### PR TITLE
ROX-21691: return and retrieve by compliance check uid

### PR DESCRIPTION
**Pull request by dashrews78**
## Description

This PR accomplishes 2 things:
1. It adds the UID of the compliance check result to results cluster section
2. It adds an endpoint to retrieve the result specifically by that ID

the compliance check UID is now included for each result in the cluster section as the UID is unique to that result and cluster.
```
{
   "scanResults" : [
      {
         "checkResults" : [
            {
               "checkId" : "xccdf_org.ssgproject.content_rule_etcd_cert_file",
               "checkName" : "ocp4-cis-etcd-cert-file",
               "clusters" : [
                  {
                     "checkUid" : "01f9f2c8-2667-4816-9259-e0651d948b1a",
                     "cluster" : {
                        "clusterId" : "1c327149-94f0-451e-8d3a-c746c2c71520",
                        "clusterName" : ""
                     },
                     "createdTime" : "2024-01-22T15:13:19Z",
                     "status" : "PASS"
                  }
               ],
               "control" : "",
               "description" : "Ensure That The etcd Client Certificate Is Correctly Set\nTo ensure the etcd service is serving TLS to clients, make sure the etcd-pod* ConfigMaps in the openshift-etcd namespace contain the following argument for the etcd binary in the etcd pod:\n\n--cert-file=/etc/kubernetes/static-pod-certs/secrets/etcd-all-[a-z]+/etcd-serving-NODE_NAME.crt\n\n. Note that the\n\n[a-z]+\n\nis being used since the directory might change between OpenShift versions.",
               "instructions" : "Run the following command:\noc get -nopenshift-etcd cm etcd-pod -oyaml | grep -E \"\\-\\-cert-file=/etc/kubernetes/static-pod-certs/secrets/etcd-all-[a-z]+/etcd-serving-NODE_NAME.crt\"\nVerify that there is a certificate configured.",
               "rationale" : "Without cryptographic integrity protections, information can be altered by unauthorized users without detection.",
               "standard" : "",
               "valuesUsed" : [],
               "warnings" : []
            },
...
         ],    
         "profileName" : "",
         "scanName" : "shrewstest6"
      }        
   ]           
} 
```

Example of a query that grabs specific check result for a cluster by check UID.
```
roxcurl "/v2/compliance/scan/results?query=Compliance%20Check%20UID:c77a7487-377b-4424-9650-dcbaddbe69f6" -u admin:THEPASS | json_pp
{
   "scanResults" : [
      {
         "checkResults" : [
            {
               "checkId" : "xccdf_org.ssgproject.content_rule_disable_ctrlaltdel_reboot",
               "checkName" : "rhcos4-moderate-master-disable-ctrlaltdel-reboot",
               "clusters" : [
                  {
                     "checkUid" : "c77a7487-377b-4424-9650-dcbaddbe69f6",
                     "cluster" : {
                        "clusterId" : "1c327149-94f0-451e-8d3a-c746c2c71520",
                        "clusterName" : ""
                     },
                     "createdTime" : "2024-01-22T15:15:07Z",
                     "status" : "FAIL"
                  }
               ],
               "control" : "",
               "description" : "Disable Ctrl-Alt-Del Reboot Activation\nBy default, SystemD will reboot the system if the Ctrl-Alt-Del key sequence is pressed.\n\nTo configure the system to ignore the Ctrl-Alt-Del key sequence from the command line instead of rebooting the system, create a MachineConfig similar to the following:\n\napiVersion: machineconfiguration.openshift.io/v1\nkind: MachineConfig\nmetadata:\n labels:\n   machineconfiguration.openshift.io/role: master\n name: 75-master-disable-ctrlaltdel-reboot\nspec:\n config:\n   ignition:\n     version: 3.1.0\n   systemd:\n     units:\n     - name: ctrl-alt-del.target\n       mask: true\nEOF\n\nThis will mask the ctrl-alt-del.target systemd target for all the nodes labeled with the \"master\" role.\n\nNote that this needs to be done for each MachineConfigPool\n\nFor more information on how to configure nodes with the Machine Config Operator see the relevant documentation ( https://docs.openshift.com/container-platform/4.6/post_installation_configuration/machine-configuration-tasks.html ).",
               "instructions" : "To ensure the system is configured to mask the Ctrl-Alt-Del sequence, Check\nthat the ctrl-alt-del.target is masked and not active with the following\ncommand:\nsudo systemctl status ctrl-alt-del.target\nThe output should indicate that the target is masked and not active. It\nmight resemble following output:\nctrl-alt-del.target\nLoaded: masked (/dev/null; bad)\nActive: inactive (dead)",
               "rationale" : "A locally logged-in user who presses Ctrl-Alt-Del, when at the console, can reboot the system. If accidentally pressed, as could happen in the case of mixed OS environment, this can create the risk of short-term loss of availability of systems due to unintentional reboot.",
               "standard" : "",
               "valuesUsed" : [],
               "warnings" : []
            }
         ],
         "profileName" : "",
         "scanName" : "shrewstest6"
      }
   ]
}
```

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
~- [ ] Evaluated and added CHANGELOG entry if required~
~- [ ] Determined and documented upgrade steps~
~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs]~(https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

Updated unit tests and added new ones for the new endpoint
Ran the cluster locally and executed the various queries to get the results and then to query specifically by ID.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: central/complianceoperator/v2/checkresults/datastore/datastore_impl_test.go [hunk 1]
```diff
@@ -319,58 +319,6 @@ func (s *complianceCheckResultDataStoreTestSuite) TestDeleteResult() {
 	s.Require().Nil(retrieveRec1)
 }
 
-func (s *complianceCheckResultDataStoreTestSuite) TestSearchCheckResults() {
```
Comment (dashrews78) [inline central/complianceoperator/v2/checkresults/datastore/datastore_impl_test.go:322]: These were duplicative as the cases were covered in the more robust tests for SAC.

**Review by theencee — APPROVED** [review]

File: central/complianceoperator/v2/checkresults/service/service_impl.go [hunk 2]
```diff
@@ -155,3 +156,20 @@ func (s *serviceImpl) GetComplianceScanResultsCount(ctx context.Context, query *
 		Count: int32(count),
 	}, nil
 }
+
+// GetComplianceScanCheckResult returns the specific result by ID
+func (s *serviceImpl) GetComplianceScanCheckResult(ctx context.Context, req *v2.ResourceByID) (*v2.ComplianceCheckResult, error) {
+	if req.GetId() == "" {
```
Comment (ksurabhi91) [inline central/complianceoperator/v2/checkresults/service/service_impl.go:162]: we should check for read access

File: central/complianceoperator/v2/checkresults/datastore/datastore_impl.go [hunk 3]
```diff
@@ -57,6 +57,11 @@ func (d *datastoreImpl) SearchComplianceCheckResults(ctx context.Context, query
 	return d.store.GetByQuery(ctx, query)
 }
 
+// GetComplianceCheckResult returns the instance of the result specified by ID
+func (d *datastoreImpl) GetComplianceCheckResult(ctx context.Context, complianceResultID string) (*storage.ComplianceOperatorCheckResultV2, bool, error) {
+	return d.store.Get(ctx, complianceResultID)
```
Comment (ksurabhi91) [inline central/complianceoperator/v2/checkresults/datastore/datastore_impl.go:62]: we should check for read access here
if ok, err := complianceOperatorSAC.ReadAllowed(ctx); err != nil {
		return nil, false,err
	} else if !ok {
		return nil ,false,sac.ErrResourceAccessDenied
	}

Comment (dashrews78) [reply central/complianceoperator/v2/checkresults/datastore/datastore_impl.go:62]: That is checked by the store

**Pull request merged**